spyder/api/widgets/menus.py

```diff
@@ -459,8 +459,9 @@ def __repr__(self):
 
     # ---- Qt methods
     # -------------------------------------------------------------------------
-    def showEvent(self, event):
+    def show(self):
         """Adjustments when the menu is shown."""
+        super().show()
```

Comment:
This call to `super().show()` can't be done at this point because the code just below this line won't be run as expected.

To test that, you can display any submenu. With your changes, it's shown like this the first time it's displayed:

![imagen](https://github.com/user-attachments/assets/e8a3e26a-0b37-4df0-be23-a1b492abe295)

As you can see, the submenu is not placed at the right vertical position with respect to the menu entries (that's what the code inside `if not self._is_shown` corrects).

Reply:
With the latest commit this is what I'm seeing on Windows:

![show_menus_submenus_win](https://github.com/user-attachments/assets/28b95d9e-cce1-4d03-b232-02657668b1af)

Reply:
Yep, it is. Thanks!